rest-server/engines/src/main/scala/com/actionml/engines/ur/URPreparator.scala

```diff
@@ -46,23 +46,30 @@ object URPreparator extends LazyLogging with SparkMongoSupport {
     val eventsRDD = PEventStore.find(
       appName = dsp.appName,
       entityType = Some("user"),
-      eventNames = Some(eventNames),
+      indicatorParams = Some(indicatorParams),
       targetEntityType = Some(Some("item")))(sc).repartition(sc.defaultParallelism)
     */
 
     // now separate the events by event name
-    val eventRDDs: Seq[(ActionID, RDD[(UserID, ItemID)])] = eventNames.map { eventName =>
+    val indicatorNames = indicatorParams.keySet
+    val eventRDDs: Seq[(String, RDD[(UserID, ItemID)])] = indicatorNames.map { indicatorName =>
+      val aliases = indicatorParams(indicatorName).aliases.getOrElse(Seq(indicatorName))
```

Comment:
I'd suggest to use Map's getOrElse method `indicatorParams(indicatorName).getOrElse(aliases, Seq(indicatorName)`